test/models/user_test.rb

```diff
@@ -95,4 +95,17 @@ def setup
     @user.reload
     assert_equal 'supervolunteer@example.com', @user.email
   end
+
+  test 'updating client mail doesnt change current user email' do
+    @superadmin.update(email: 'superadmin@example.com')
+    @superadmin.reload
+    @client = create :user, email: 'client@example.com'
+    login_as(@superadmin)
+    @client.profile.contact.update(primary_email: 'superclient@example.com')
+    @client.reload
+
+    assert_equal 'superclient@example.com', @client.profile.contact.primary_email
+    assert_not_equal 'superclient@example.com', @superadmin.email
```

Comment:
You can remove this line since you have an `assert_equal ..., @superadmin.email` below :)

Reply:
Actually you can remove `@superadmin` completely from this test, since you're already testing that case in the existing test above and only really need to verify the behaviour for clients here :)